gtda/plotting/persistence_diagrams.py

```diff
@@ -37,23 +37,26 @@ def plot_diagram(diagram, homology_dimensions=None, plotly_params=None):
         homology_dimensions = np.unique(diagram[:, 2])
 
     diagram_no_dims = diagram[:, :2]
-    max_birth, max_death = np.max(
-        np.where(np.isposinf(diagram_no_dims), -np.inf, diagram_no_dims),
-        axis=0
+    max_val_display = np.max(
```

Comment:
Taking the global maxima and minima makes the code ready for extended persistence.